tequilapi/endpoints/service.go

```diff
@@ -189,58 +222,82 @@ func (se *ServiceEndpoint) ServiceStart(resp http.ResponseWriter, req *http.Requ
 //     schema:
 //       "$ref": "#/definitions/ErrorMessageDTO"
 func (se *ServiceEndpoint) ServiceStop(resp http.ResponseWriter, _ *http.Request, params httprouter.Params) {
-	if params.ByName("id") != "6ba7b810-9dad-11d1-80b4-00c04fd430c8" && params.ByName("id") != "7037d6f6-1b92-4513-8cb4-8fd1a944b996" {
+	id := service.ID(params.ByName("id"))
+	service := se.serviceManager.Service(id)
+
+	if service == nil {
 		utils.SendErrorMessage(resp, "Service not found", http.StatusNotFound)
 		return
 	}
-	se.serviceState = service.NotRunning
 
+	if err := se.serviceManager.Stop(id); err != nil {
+		utils.SendError(resp, err, http.StatusInternalServerError)
+		return
+	}
 	resp.WriteHeader(http.StatusAccepted)
 }
 
+func (se *ServiceEndpoint) isAlreadyRunning(sr serviceRequest) bool {
+	for _, instance := range se.serviceManager.List() {
+		proposal := instance.Proposal()
+		if proposal.ProviderID == sr.ProviderID && proposal.ServiceType == sr.ServiceType {
+			return true
+		}
+	}
+	return false
+}
+
 // AddRoutesForService adds service routes to given router
-func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManager) {
-	serviceEndpoint := NewServiceEndpoint(serviceManager)
+func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManager, optionsParser map[string]func(json.RawMessage) (service.Options, error)) {
+	serviceEndpoint := NewServiceEndpoint(serviceManager, optionsParser)
 
 	router.GET("/services", serviceEndpoint.ServiceList)
 	router.POST("/services", serviceEndpoint.ServiceStart)
 	router.GET("/services/:id", serviceEndpoint.ServiceGet)
 	router.DELETE("/services/:id", serviceEndpoint.ServiceStop)
 }
 
-func toServiceRequest(req *http.Request) (*serviceRequest, error) {
-	var serviceRequest = serviceRequest{}
-	err := json.NewDecoder(req.Body).Decode(&serviceRequest)
-
-	return &serviceRequest, err
+func toServiceRequest(req *http.Request) (sr serviceRequest, err error) {
+	err = json.NewDecoder(req.Body).Decode(&sr)
+	return sr, err
 }
 
-func toServiceInfoResponse(state service.State) serviceInfo {
+func toServiceInfoResponse(id service.ID, instance *service.Instance) serviceInfo {
 	return serviceInfo{
-		Status: string(state),
+		ID:       string(id),
+		Status:   string(service.Running),
+		Proposal: proposalToRes(instance.Proposal()),
 	}
 }
 
-func toServiceListResponse(state service.State) serviceList {
-	return serviceList{
-		{ID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", Status: string(state)},
-		{ID: "7037d6f6-1b92-4513-8cb4-8fd1a944b996", Status: string(state)},
+func toServiceListResponse(instances map[service.ID]*service.Instance) serviceList {
+	res := make([]serviceInfo, 0)
+	for id, instance := range instances {
+		res = append(res, serviceInfo{
```

Comment:
You could reuse `toServiceInfoResponse()` here

Reply:
Done.